WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -167,15 +216,44 @@ public void onSitePluginUpdated(OnSitePluginUpdated event) {
                     + event.error.message);
             return;
         }
-        if (event.plugin != null) {
-            mPlugin = event.plugin;
-            refreshStates();
+        mPlugin = mPluginStore.getSitePluginByName(mSite, mPlugin.getName());
+        refreshViews();
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginInfoChanged(PluginStore.OnPluginInfoChanged event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "An error occurred while fetching the plugin info with type: "
+                    + event.error.type);
+            return;
+        }
+        if (event.pluginInfo != null && mPlugin.getSlug().equals(event.pluginInfo.getSlug())) {
+            mPluginInfo = event.pluginInfo;
+            refreshPluginVersionViews();
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSitePluginVersionUpdated(PluginStore.OnSitePluginVersionUpdated event) {
+        if (event.isError()) {
+            AppLog.e(AppLog.T.API, "An error occurred while updating the plugin version with type: "
```

Comment:
Should we alert the user that an error occurred here?

Reply:
Yes, as mentioned in the PR description, we'll use a snackbar to show that update has started, successful and there is an error. I am also adding an activity indicator for where the update button is, so the user can't tap it again while an update is running.

> Another PR is incoming to polish the update feature a bit with Activity indicator and a snackbar to let the user know what's happening.